command/server/server.go

```diff
@@ -238,6 +249,45 @@ func isConfigFileSpecified(cmd *cobra.Command) bool {
 func runCommand(cmd *cobra.Command, _ []string) {
 	outputter := command.InitializeOutputter(cmd)
 
+	log.Println("Main process run isDaemon:", params.isDaemon)
+
+	// Launch daemons
+	if params.isDaemon {
+		// First time, err is not empty
+		_, err := strconv.Atoi(os.Getenv(daemon.EnvName))
+		if err != nil {
+			input, err := gopass.GetPasswdPrompt("Enter ValidatorKey:", true, os.Stdin, os.Stdout)
+			if err != nil {
+				log.Println("Parent process ", os.Getpid(), " passwd prompt err:", err)
+			}
+
+			params.validatorKey = string(input)
+			log.Println("Parent process ", os.Getpid(), " passwd prompt, ValidatorKey:", len(params.validatorKey))
+		} else {
+			data, err := ioutil.ReadAll(os.Stdin)
+			if err != nil {
+				log.Println("Child process ", os.Getpid(), " read pipe data err: ", err)
+			} else {
+				log.Println("Child process ", os.Getpid(), " read pipe data: ", len(string(data)))
+				params.validatorKey = string(data)
+			}
+		}
+
+		// Create a daemon object
+		logFile := "daemon.log"
+		newDaemon := daemon.NewDaemon(logFile)
+		newDaemon.MaxCount = 5
+		newDaemon.ValidatorKey = params.validatorKey
+
+		// Execute daemon mode
+		newDaemon.Run()
+
+		//当 *d = true 时以下代码只有最终子进程会执行, 主进程和守护进程都不会执行
+		log.Println("Child process ", os.Getpid(), "start...")
+		log.Println("Child process ", os.Getpid(), "isDaemon: ", params.isDaemon)
+		log.Println("Child process ", os.Getpid(), "ValidatorKey: ", len(params.validatorKey))
```

Comment:
Better not print out this private key, since it is logging in file.
Could print out the public key, for user confirmation.

Reply:
😅 My mistake.
The `validatorKey` would only print out the length.